src/cmd/root.go

```diff
@@ -136,7 +137,12 @@ func preRun(cmd *cobra.Command, args []string) error {
 	logrus.Debugf("Running as real user ID %s", currentUser.Uid)
 	logrus.Debugf("Resolved absolute path to the executable as %s", executable)
 
-	if !utils.IsInsideContainer() {
+	if cmdName, completionCmdName := cmd.Name(), completionCmd.Name(); cmdName == completionCmdName {
+		logrus.Debugf("Migration not needed: command %s doesn't need it", cmdName)
+		migrationNeeded = false
+	}
+
+	if migrationNeeded && !utils.IsInsideContainer() {
```

Comment:
This code that the `migrationNeeded` variable is guarding isn't actually related to migrating to newer Podman versions at all.  So, the code becomes a bit misleading.

It might be better to wrap `utils.ValidateSubIDRanges` in (another) convenience function that encapsulates all these details of whether we are generating the shell completions or inside a container or running as root.

Reply:
https://github.com/containers/toolbox/pull/1252